File: backend/data/mock_data.py
"""
mock_data.py
────────────
Single source of truth for all mock data.
Mirrors frontend/lib/mock-data.ts exactly so both stacks stay in sync.
"""
from datetime import datetime, timedelta
import random

# ── Patients ────────────────────────────────────────────────────────────────
PATIENTS = [
    {"id": "p1", "name": "Jane Doe",        "age": 28, "pregnancy_week": 28, "risk_level": "low",    "status": "active"},
    {"id": "p2", "name": "Priya Sharma",    "age": 32, "pregnancy_week": 24, "risk_level": "medium", "status": "active"},
    {"id": "p3", "name": "Amara Okonkwo",   "age": 26, "pregnancy_week": 20, "risk_level": "high",   "status": "active"},
    {"id": "p4", "name": "Sarah Johnson",   "age": 30, "pregnancy_week": 32, "risk_level": "low",    "status": "pending"},
    {"id": "p5", "name": "Maria Garcia",    "age": 27, "pregnancy_week": 16, "risk_level": "low",    "status": "active"},
]

# ── Vitals ───────────────────────────────────────────────────────────────────
VITALS = {
    "blood_glucose":      7.2,
    "blood_pressure_sys": 130,
    "blood_pressure_dia": 85,
    "weight":             68.5,
    "weight_change":      1.2,
}

# ── 30-day vitals history ────────────────────────────────────────────────────
random.seed(42)
VITALS_HISTORY = [
    {
        "day":      i + 1,
        "glucose":  round(6.8 + random.random() * 2, 2),
        "systolic": round(120 + random.random() * 20, 1),
        "diastolic":round(75  + random.random() * 20, 1),
    }
    for i in range(30)
]

# ── Meals ────────────────────────────────────────────────────────────────────
MEALS = [
    {"id": "m1", "name": "Ragi Porridge with Jaggery",    "type": "breakfast", "calories": 280, "ingredients": ["Ragi flour", "Jaggery", "Ghee", "Cardamom"],                       "serving_size": "200"},
    {"id": "m2", "name": "Spinach and Lentil Dal",         "type": "lunch",     "calories": 350, "ingredients": ["Red lentils", "Spinach", "Turmeric", "Cumin"],                     "serving_size": "250"},
    {"id": "m3", "name": "Red Rice with Vegetables",       "type": "lunch",     "calories": 380, "ingredients": ["Red rice", "Carrots", "Beans", "Peas", "Coconut oil"],             "serving_size": "280"},
    {"id": "m4", "name": "Banana and Almond Smoothie",     "type": "snack",     "calories": 210, "ingredients": ["Banana", "Almonds", "Milk", "Honey"],                             "serving_size": "200"},
    {"id": "m5", "name": "Millet Upma with Vegetables",    "type": "dinner",    "calories": 320, "ingredients": ["Millet", "Carrot", "Peas", "Mustard seeds", "Onion"],             "serving_size": "250"},
]

# ── Meal Orders ───────────────────────────────────────────────────────────────
MEAL_ORDERS = [
    {"id": "o1", "patient_id": "p1", "patient_name": "Jane Doe",      "meal_type": "Ragi Porridge",  "due_time": "08:00 AM", "status": "pending",   "quantity": 1},
    {"id": "o2", "patient_id": "p1", "patient_name": "Jane Doe",      "meal_type": "Spinach Dal",    "due_time": "12:30 PM", "status": "preparing", "quantity": 1},
    {"id": "o3", "patient_id": "p2", "patient_name": "Priya Sharma",  "meal_type": "Red Rice Bowl",  "due_time": "12:30 PM", "status": "ready",     "quantity": 2},
    {"id": "o4", "patient_id": "p3", "patient_name": "Amara Okonkwo", "meal_type": "Millet Upma",    "due_time": "06:00 PM", "status": "pending",   "quantity": 1},
]

# ── Alerts ───────────────────────────────────────────────────────────────────
now = datetime.now()
ALERTS = [
    {"id": "a1", "patient_id": "p1", "type": "critical", "message": "Blood glucose spike detected: 9.1 mmol/L",    "timestamp": now,                        "resolved": False},
    {"id": "a2", "patient_id": "p3", "type": "warning",  "message": "Missing weight entry for today",              "timestamp": now - timedelta(hours=1),   "resolved": False},
    {"id": "a3", "patient_id": "p2", "type": "warning",  "message": "Blood pressure elevated: 145/92 mmHg",        "timestamp": now - timedelta(hours=2),   "resolved": True},
]

# ── AI Meal Plan ──────────────────────────────────────────────────────────────
AI_MEAL_PLAN = {
    "id":              "plan1",
    "patient_id":      "p1",
    "status":          "pending",
    "generated_at":    now,
    "approved_at":     None,
    "meals": [
        {"time": "8:00 AM",  "meal": MEALS[0], "calories": 280},
        {"time": "11:00 AM", "meal": MEALS[3], "calories": 210},
        {"time": "1:00 PM",  "meal": MEALS[1], "calories": 350},
        {"time": "4:00 PM",  "meal": MEALS[3], "calories": 210},
        {"time": "7:00 PM",  "meal": MEALS[4], "calories": 320},
    ],
    "total_calories":  1370,
    "nutrition_score": 8.5,
}

# ── Pending Approval Plans ────────────────────────────────────────────────────
PENDING_PLANS = [
    {"id": "ap1", "patient_name": "Jane Doe",      "meal_count": 5, "total_calories": 1370, "generated_at": now,                        "status": "pending"},
    {"id": "ap2", "patient_name": "Priya Sharma",  "meal_count": 4, "total_calories": 1150, "generated_at": now - timedelta(hours=2),   "status": "pending"},
    {"id": "ap3", "patient_name": "Amara Okonkwo", "meal_count": 6, "total_calories": 1520, "generated_at": now - timedelta(hours=4),   "status": "pending"},
]

# ── Nutrition macros ──────────────────────────────────────────────────────────
NUTRITION_MACROS = {"carbs": 280, "protein": 85, "fats": 65, "fiber": 35}

# ── Overview KPI stats ────────────────────────────────────────────────────────
OVERVIEW_STATS = {
    "total_patients":     len(PATIENTS),
    "active_orders":      len(MEAL_ORDERS),
    "pending_approvals":  len(PENDING_PLANS),
    "critical_alerts":    sum(1 for a in ALERTS if a["type"] == "critical" and not a["resolved"]),
}

File: frontend/lib/mock-data.ts
// Mock Data for Healthcare Dashboard

export interface Patient {
  id: string;
  name: string;
  age: number;
  pregnancyWeek: number;
  riskLevel: 'low' | 'medium' | 'high';
  status: 'active' | 'pending' | 'completed';
}

export interface Vitals {
  bloodGlucose: number;
  bloodPressureSys: number;
  bloodPressureDia: number;
  weight: number;
  weightChange: number;
}

export interface Meal {
  id: string;
  name: string;
  type: 'breakfast' | 'lunch' | 'dinner' | 'snack';
  calories: number;
  ingredients: string[];
  servingSize: string;
}

export interface MealOrder {
  id: string;
  patientId: string;
  patientName: string;
  mealType: string;
  dueTime: string;
  status: 'pending' | 'preparing' | 'ready' | 'delivered';
  quantity: number;
}

export interface ChatMessage {
  id: string;
  sender: 'user' | 'bot' | 'system';
  content: string;
  timestamp: Date;
}

export interface Alert {
  id: string;
  patientId: string;
  type: 'critical' | 'warning' | 'info';
  message: string;
  timestamp: Date;
  resolved: boolean;
}

// Mock Patients
export const MOCK_PATIENTS: Patient[] = [
  {
    id: 'p1',
    name: 'Jane Doe',
    age: 28,
    pregnancyWeek: 28,
    riskLevel: 'low',
    status: 'active',
  },
  {
    id: 'p2',
    name: 'Priya Sharma',
    age: 32,
    pregnancyWeek: 24,
    riskLevel: 'medium',
    status: 'active',
  },
  {
    id: 'p3',
    name: 'Amara Okonkwo',
    age: 26,
    pregnancyWeek: 20,
    riskLevel: 'high',
    status: 'active',
  },
  {
    id: 'p4',
    name: 'Sarah Johnson',
    age: 30,
    pregnancyWeek: 32,
    riskLevel: 'low',
    status: 'pending',
  },
  {
    id: 'p5',
    name: 'Maria Garcia',
    age: 27,
    pregnancyWeek: 16,
    riskLevel: 'low',
    status: 'active',
  },
];

// Mock Vitals for Primary Patient
export const MOCK_VITALS: Vitals = {
  bloodGlucose: 7.2,
  bloodPressureSys: 130,
  bloodPressureDia: 85,
  weight: 68.5,
  weightChange: 1.2,
};

// Mock Historical Vitals Data (30 days)
export const MOCK_VITALS_HISTORY = Array.from({ length: 30 }, (_, i) => ({
  day: i + 1,
  glucose: 6.8 + Math.random() * 2,
  systolic: 120 + Math.random() * 20,
  diastolic: 75 + Math.random() * 20,
}));

// Mock Meals
export const MOCK_MEALS: Meal[] = [
  {
    id: 'm1',
    name: 'Ragi Porridge with Jaggery',
    type: 'breakfast',
    calories: 280,
    ingredients: ['Ragi flour', 'Jaggery', 'Ghee', 'Cardamom'],
    servingSize: '200g',
  },
  {
    id: 'm2',
    name: 'Spinach and Lentil Dal',
    type: 'lunch',
    calories: 350,
    ingredients: ['Red lentils', 'Spinach', 'Turmeric', 'Cumin'],
    servingSize: '250g',
  },
  {
    id: 'm3',
    name: 'Red Rice with Vegetables',
    type: 'lunch',
    calories: 380,
    ingredients: ['Red rice', 'Carrots', 'Beans', 'Peas', 'Coconut oil'],
    servingSize: '280g',
  },
  {
    id: 'm4',
    name: 'Banana and Almond Smoothie',
    type: 'snack',
    calories: 210,
    ingredients: ['Banana', 'Almonds', 'Milk', 'Honey'],
    servingSize: '200ml',
  },
  {
    id: 'm5',
    name: 'Millet Upma with Vegetables',
    type: 'dinner',
    calories: 320,
    ingredients: ['Millet', 'Carrot', 'Peas', 'Mustard seeds', 'Onion'],
    servingSize: '250g',
  },
];

// Mock Meal Orders
export const MOCK_MEAL_ORDERS: MealOrder[] = [
  {
    id: 'o1',
    patientId: 'p1',
    patientName: 'Jane Doe',
    mealType: 'Ragi Porridge',
    dueTime: '08:00 AM',
    status: 'pending',
    quantity: 1,
  },
  {
    id: 'o2',
    patientId: 'p1',
    patientName: 'Jane Doe',
    mealType: 'Spinach Dal',
    dueTime: '12:30 PM',
    status: 'preparing',
    quantity: 1,
  },
  {
    id: 'o3',
    patientId: 'p2',
    patientName: 'Priya Sharma',
    mealType: 'Red Rice Bowl',
    dueTime: '12:30 PM',
    status: 'ready',
    quantity: 2,
  },
  {
    id: 'o4',
    patientId: 'p3',
    patientName: 'Amara Okonkwo',
    mealType: 'Millet Upma',
    dueTime: '06:00 PM',
    status: 'pending',
    quantity: 1,
  },
];

// Mock Chat Messages
export const MOCK_NUTRITIONIST_MESSAGES: ChatMessage[] = [
  {
    id: 'c1',
    sender: 'system',
    content: 'Welcome to your nutritionist chat. I am here to help you with any dietary concerns.',
    timestamp: new Date(Date.now() - 3600000),
  },
  {
    id: 'c2',
    sender: 'user',
    content: 'I have been feeling nauseous after the afternoon meals.',
    timestamp: new Date(Date.now() - 3000000),
  },
  {
    id: 'c3',
    sender: 'bot',
    content: 'I recommend reducing portion sizes for lunch and including ginger tea 30 minutes before meals.',
    timestamp: new Date(Date.now() - 2700000),
  },
];

export const MOCK_CHEF_MESSAGES: ChatMessage[] = [
  {
    id: 'ch1',
    sender: 'system',
    content: 'Chef Assistant: Ready to help with recipe guidance and cooking tips.',
    timestamp: new Date(Date.now() - 7200000),
  },
  {
    id: 'ch2',
    sender: 'user',
    content: 'How should I adjust the spinach dal for bulk cooking?',
    timestamp: new Date(Date.now() - 6000000),
  },
  {
    id: 'ch3',
    sender: 'bot',
    content: 'For bulk orders, scale the recipe 1:3 ratio. Increase cooking time by 5-10 minutes and stir occasionally.',
    timestamp: new Date(Date.now() - 5400000),
  },
];

// Mock Alerts
export const MOCK_ALERTS: Alert[] = [
  {
    id: 'a1',
    patientId: 'p1',
    type: 'critical',
    message: 'Blood glucose spike detected: 9.1 mmol/L',
    timestamp: new Date(),
    resolved: false,
  },
  {
    id: 'a2',
    patientId: 'p3',
    type: 'warning',
    message: 'Missing weight entry for today',
    timestamp: new Date(Date.now() - 3600000),
    resolved: false,
  },
  {
    id: 'a3',
    patientId: 'p2',
    type: 'warning',
    message: 'Blood pressure elevated: 145/92 mmHg',
    timestamp: new Date(Date.now() - 7200000),
    resolved: true,
  },
];

// Mock AI Meal Plans
export const MOCK_AI_MEAL_PLAN = {
  id: 'plan1',
  patientId: 'p1',
  status: 'pending' as const,
  generatedAt: new Date(),
  approvedAt: null,
  meals: [
    { time: '8:00 AM', meal: MOCK_MEALS[0], calories: 280 },
    { time: '11:00 AM', meal: MOCK_MEALS[3], calories: 210 },
    { time: '1:00 PM', meal: MOCK_MEALS[1], calories: 350 },
    { time: '4:00 PM', meal: MOCK_MEALS[3], calories: 210 },
    { time: '7:00 PM', meal: MOCK_MEALS[4], calories: 320 },
  ],
  totalCalories: 1370,
  nutritionScore: 8.5,
};

export const MOCK_PENDING_PLANS = [
  {
    id: 'ap1',
    patientName: 'Jane Doe',
    mealCount: 5,
    totalCalories: 1370,
    generatedAt: new Date(),
    status: 'pending',
  },
  {
    id: 'ap2',
    patientName: 'Priya Sharma',
    mealCount: 4,
    totalCalories: 1150,
    generatedAt: new Date(Date.now() - 7200000),
    status: 'pending',
  },
  {
    id: 'ap3',
    patientName: 'Amara Okonkwo',
    mealCount: 6,
    totalCalories: 1520,
    generatedAt: new Date(Date.now() - 14400000),
    status: 'pending',
  },
];

// ─────────────────────────────────────────────────────────────────────────────
// API Endpoints
// Centralise all backend route paths here so components never hardcode strings.
// Sync contract: keep in alignment with
//   frontend/app/api/chat/nutritionist/route.ts
//   frontend/app/api/chat/chef/route.ts
// ─────────────────────────────────────────────────────────────────────────────
export const API_ENDPOINTS = {
  /** Patient Portal — Dr. NutriCare AI (gpt-4o, temp 0.5) */
  chatNutritionist: '/api/chat/nutritionist',
  /** Kitchen Command — ChefAid AI (gpt-4o, temp 0.4) */
  chatChef: '/api/chat/chef',
} as const;

// ─────────────────────────────────────────────────────────────────────────────
// Brand / Design Tokens  (mirrors backend/utils/styles.py)
// Use these in components instead of raw hex strings so colour changes
// only need to happen in ONE place.
// ─────────────────────────────────────────────────────────────────────────────
export const BRAND_COLORS = {
  sage:    '#6B8E6F',   // primary brand green
  teal:    '#20B2AA',   // safe / approved state
  amber:   '#F59E0B',   // warning / pending state
  crimson: '#DC2626',   // critical / urgent state
} as const;
